File: Go/167. Two Sum II - Input Array Is Sorted.go
func twoSum(numbers []int, target int) []int {
    a, b := len(numbers) - 1, 0
    for true{
        if numbers[a] + numbers[b] > target {
            a--
        } else if numbers[a] + numbers[b] < target {
            b++
        } else {
            return []int{b + 1, a + 1}
        }
    }
    return []int{b + 1, a + 1}
}

File: Swift/167. Two Sum II - Input Array Is Sorted.swift
class Solution {
    func twoSum(_ numbers: [Int], _ target: Int) -> [Int] {
        var a = numbers.count - 1, b = 0;
        while(numbers[a] + numbers[b] != target){
            if(numbers[a] + numbers[b] > target){ a -= 1; }
            else{ b += 1; }
        }
        return [b + 1, a + 1];
    }
}
